lightningd/gossip/gossip.c

```diff
@@ -105,6 +105,47 @@ static struct peer *setup_new_peer(struct daemon *daemon, const u8 *msg)
 	return peer;
 }
 
+static struct io_plan *owner_msg_in(struct io_conn *conn,
+				    struct daemon_conn *dc);
+static struct io_plan *nonlocal_dump_gossip(struct io_conn *conn,
+					    struct daemon_conn *dc);
+
+/* When a peer is to be owned by another daemon, we create a socket
+ * pair to send/receive gossip from it */
+static void send_peer_with_fds(struct peer *peer, const u8 *msg)
+{
+	int fds[2];
+	u8 *out;
+
+	if (socketpair(AF_LOCAL, SOCK_STREAM, 0, fds) != 0) {
+		out = towire_gossipstatus_peer_failed(msg,
+				peer->unique_id,
+				(u8 *)tal_fmt(msg,
+					      "Failed to create socketpair: %s",
+					      strerror(errno)));
+		daemon_conn_send(&peer->daemon->master, take(out));
+
+		/* FIXME: Send error to peer? */
+		/* Peer will be freed when caller closes conn. */
+		return;
+	}
+
+	/* Now we talk to socket to get to peer's owner daemon. */
+	peer->local = false;
+	daemon_conn_init(peer, &peer->owner_conn, fds[0], owner_msg_in);
+	peer->owner_conn.msg_queue_cleared_cb = nonlocal_dump_gossip;
+
+	/* Peer stays around, even though we're going to free conn. */
+	tal_steal(peer->daemon, peer);
```

Comment:
I was wondering if we could `tal_steal` onto our end of the connection so that the peer gets cleaned up upon connection close, would that work?